src/coreclr/vm/codeman.cpp

```diff
@@ -4339,8 +4339,25 @@ BOOL InterpreterJitManager::JitCodeToMethodInfo(
 
 TADDR InterpreterJitManager::GetFuncletStartAddress(EECodeInfo * pCodeInfo)
 {
-    // Interpreter-TODO: Verify that this is correct
-    return pCodeInfo->GetCodeAddress() - pCodeInfo->GetRelOffset();
+    EH_CLAUSE_ENUMERATOR    EnumState;
+    unsigned                EHCount;
+
+    IJitManager *pJitMan = pCodeInfo->GetJitManager();
+    EHCount = pJitMan->InitializeEHEnumeration(pCodeInfo->GetMethodToken(), &EnumState);
+    DWORD relOffset = pCodeInfo->GetRelOffset();
+
+    for (unsigned i = 0; i < EHCount; i++)
+    {
+        EE_ILEXCEPTION_CLAUSE EHClause;
+        pJitMan->GetNextEHClause(&EnumState, &EHClause);
+
+        if (EHClause.HandlerStartPC <= relOffset && relOffset < EHClause.HandlerEndPC)
```

Comment:
Does this work for filters?

Reply:
I am not sure whether you wanted to push this into this PR since there is no matching logic in the JIT, so it is hard to tell whether it is correct.

Reply:
Yes, it works for filters too. The interpreter compiler sets the `HandlerEndPC` even for filters.

> I am not sure whether you wanted to push this into this PR since there is no matching logic in the JIT, so it is hard to tell whether it is correct.

Makes sense, let's make it part of Andrew's part of the change instead.

Reply:
Hmm, actually, even JIT sets the `HandlerEndPC` for filters

Reply:
The problem that I see is that the filter funclet is a separate funclet that is not included in the (HandlerStartPC, HandledEndPC) range, with the JIT at least. I assume that you want to use the same encoding rules for interpreter as well.

I have done a quick test.
```
try
{
    Console.WriteLine("Try");
}
catch (Exception e) when (args.Length > 10)
{
    Console.WriteLine("Catch");
}
```
Produces 
```
<img width="155" alt="image" src="https://github.com/user-attachments/assets/820b6b67-e846-4cb8-bb17-f780d911ee49" />
```
Notice that FilterOffset is not included (HandlerStartPC, HandledEndPC).

Reply:
Ah, I am sorry, you are right, the ranges I am scanning contain the catch handler for the filter, not the filter itself. That's the same thing for the interpreter then.

Reply:
We will need to figure out some other mechanism for getting the end address of filter funclets. The JIT uses unwind info stuff, which we don't have.

Reply:
You can add a rule that the filter funclet must immediately precede the handler. (This rule would apply to interpreter JIT only.)

Reply:
Thank you, that sounds like a reasonable solution.